### DISCUSSION TITLE
Using GPU on google colab

### DISCUSSION THREAD

**Discussion by saksonita** [ourownstory/neural_prophet#1409] [Q&A - get help using NeuralProphet]
I am stuck with utilizing the GPU to run the neural prophet on the google colab. I tried to configure the accelerator to use the GPU but it always use the CPU. I need help on that. 

Looking forward to your responding. 

<img width="1668" alt="image" src="https://github.com/ourownstory/neural_prophet/assets/12293971/3a6802a8-bb8d-4046-8dbb-01f65237c2a7">

**Comment by leoniewgnr** [comment]
@saksonita mmh this is weird...
This normally works for, but I'm also not using google colab. Did you try it on another GPU (like azure or local) if the problem still exists there?

**Comment by saksonita** [comment]
Thank for your response. I also tried with my local machine (GTX 1080Ti).
But I still encounter the same problem.

[image: image.png]

[image: image.png]

On Tue, Aug 29, 2023 at 1:45 AM leoniewgnr ***@***.***> wrote:

> @saksonita <https://github.com/saksonita> mmh this is weird...
> This normally works for, but I'm also not using google colab. Did you try
> it on another GPU (like azure or local) if the problem still exists there?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ourownstory/neural_prophet/discussions/1409#discussioncomment-6844709>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC5ZOU256BB2JMPAH3VMICLXXTDJPANCNFSM6AAAAAA36D7BHY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Yours sincerely,

Khoeurn Saksonita

**Comment by joejoe2** [comment]
@leoniewgnr I have same problem. I found the the model attributes are weired
```
print(model.trainer.accelerator)
print(model.model.device)
```

```
<pytorch_lightning.accelerators.cuda.CUDAAccelerator object at 0x7aebef2138e0>
cpu
```

Does it mean the model is train on gpu ? But I do not see any gpu ram usage.

**Comment by ksraghuram** [comment]
Sharing what worked for me. 
while creating the model, use         
trainer_config={
        'accelerator': 'gpu',
        'devices': [0] ## put whatever device indexing your current gpu has.
    }
  instead of accelerator = 'gpu'. 
  But do let me know if this impacts the speed of training and prediction. For me, it's the same.
